en/ros2/px4_ros2_control_interface.md

```diff
@@ -0,0 +1,455 @@
+  It is a state machine that can activate modes, and wait for their completion.
+- It can only do so while it is in charge.
+  For that, an executor has exactly one _owned mode_ (and a mode can be owned by at most one executor).
+  This mode serves as activation for the executor: when the user selects the mode, the owning executor gets activated and can select any mode.
+  It stays in charge until the user switches modes (by RC or from a GCS), or a failsafe triggers a mode switch.
+  Should the failsafe clear, the executor gets reactivated.
+- This allows multiple executors to coexist.
+- Executors cannot activate other executors.
+- Within the library, a mode executor is always implemented in combination with a custom mode.
+
+:::note
+
+- These definitions guarantee that a user can take away control from a custom mode or executor at any point in time by commanding a mode switch through RC or a GCS.
+- A mode executor is transparent to the user.
+  It gets indirectly selected and activated through the owning mode, and thus the mode should be named accordingly.
+
+:::
+
+#### Configuration Overrides
+
+Both modes and executors can define configuration overrides, allowing customisation of certain behaviors while the mode or executor is active.
+
+These are currently implemented:
+
+- _Disabling auto-disarm_.
+  This permits landing and then taking off again (e.g. to release a payload).
+- _Ability to defer non-essential failsafes_.
+  This allows an executor to run an action without being interrupted by non-critical failsafe.
+  For example, ignoring a low-battery failsafe so that a winch operation can complete.
+
+### Comparison to Offboard Control
+
+The above concepts provide a number of advantages over traditional [offboard control](../ros/offboard_control.md):
+
+- Multiple nodes or applications can coexist and even run at the same time.
+  But only one node can _control the vehicle_ at a given time, and this node is well defined.
+- Modes have a distinct name and be displayed/selected in the GCS.
+- Modes are integrated with the failsafe state machine and arming checks.
+- The setpoint types that can be sent are well defined.
+- ROS 2 modes can replace flight controller internal modes (such as [Return mode](../flight_modes/return.md)).
+
+## Installation and First Test
+
+The following steps are required to get started:
+
+1. Make sure you have a working [ROS 2 setup](../ros/ros2_comm.md), with [`px4_msgs`](https://github.com/PX4/px4_msgs) in the ROS 2 workspace.
+2. Clone the repository into the workspace:
+
+   ```sh
+   cd $ros_workspace/src
+   git clone --recursive https://github.com/Auterion/px4-ros2-interface-lib
+   ```
+
+   :::note
+   To ensure compatibility, use the latest _main_ branches for PX4, _px4_msgs_ and the library.
+   See also [here](https://github.com/Auterion/px4-ros2-interface-lib#compatibility-with-px4).
+   :::
+
+3. Build the workspace:
+
+   ```sh
+   cd ..
+   colcon build
+   source install/setup.bash
+   ```
```

Comment:
FWIW all this is now duplicated in three places. I think that's OK, but if we end up with too many of these we might instead link up for this.